docs/interactions.rst

```diff
@@ -0,0 +1,126 @@
+:orphan:
+
+.. currentmodule:: nextcord
+
+.. _interactions:
+   
+Slash Commands
+==============
+
+Since Discord has added interaction commands a feature with many possibilities, we at Nextcord have decided to add them to our fleet of features.
+
+This doc will explain the innerworkings and how to use interaction commmands.
+
+We suggest you learn how to make regular commands before looking through here, we suggest looking at the :doc:`quickstart`
+
+How To Make A Simple Slash Command
+----------------------------------
+
+This right here is a simple ping pong command made with Nextcord's slash feature.
+
+.. code-block:: python3
+
+    @client.slash_command()
+    async def ping(interaction):
+        await interaction.response.send_message("Pong!")
+        
+The way it works is that you use the :meth:`~Client.slash_command` function to interact with discord's application commands endpoint.
+
+``guild_ids`` is used to limit the guilds that the slash command is available to. This is also useful for testing, as global slash commands can take up to an hour to register.
+
+Example:
+
+.. code-block:: python3
+      
+    @client.slash_command(guild_ids=[id1, id2])
+
+How To Use Sub-Commands
+-----------------------
+
+The way sub-commands work is that you make a normal slash command that will never be called, and then make the sub-commands and have them do the work of real slash commands. There is no difference with slash commands and sub-commands. The only thing you will need to change is functions.
+
+As shown in the demonstration below you make a main slash command or a dummy slash command and build sub-commands off it.
+
+.. code-block:: python3
+
+        @client.slash_command(guild_ids=[...])  # Making the command and limiting the guilds
+        async def my_main_command(interaction):  
+            ...
+
+
+        @my_main_command.subcommand()  # Identifying The Sub-Command
+        async def subcommand_one(interaction):  # Making The Sub Command Name And Passing Through Interaction
+        await interaction.response.send_message(
+            "This is subcommand 1!"
+            )  # Sending A Response
+
+
+        # Identifying The Sub-Command And Adding A Descripton
+        @my_main_command.subcommand(description="Aha! Another subcommand")  
+        async def subcommand_two(interaction: Interaction):  # Passing in interaction
+
+        await interaction.response.send_message(
+            "This is subcommand 2!"
+            )       # Responding With The Args/Fields
+        
+Fields And Requirements
```

Comment:
Nothing in the following section is named a 'field', I'm an advanced user and I have a difficult time trying to understand what is meant by half of this. I'd suggest rewriting this portion to reflect the naming in the library itself.